client/src/googleAuth/signIn.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect } from 'react';
+import { notification, Carousel } from 'antd';
+import { renderButton, checkSignedIn } from './authUtils';
+import DayVisitsReport from '../components/Charts/dayVisitsReport';
+import PageviewsReport from '../components/Charts/pageviewReport';
+import './style.css';
+
+function GoogleLogin() {
+  const [isSignedIn, setIsSignedIn] = useState(false);
+
+  const updateSignin = (signedIn) => {
+    setIsSignedIn(signedIn);
+    if (!signedIn) {
+      renderButton();
+    }
+  };
+
+  const init = () => {
+    checkSignedIn()
+      .then((signedIn) => {
+        updateSignin(signedIn);
+        window.gapi.auth2.getAuthInstance().isSignedIn.listen(updateSignin);
+      })
+      .catch((error) => {
+        const { message } = error;
+        notification.error({
+          message: 'Error',
+          description: message,
+        });
+      });
```

Comment:
`checkSignedIn` function returns a promise, so you can  use `Async/await `as we agreed to work on in our project .
```suggestion
 try {
      const signedIn = await checkSignedIn();
      updateSignin(signedIn);
      window.gapi.auth2.getAuthInstance().isSignedIn.listen(updateSignin);
    } catch (error) {
      const { message } = error;
      notification.error({
        message: 'Error',
        description: message,
      });
    }
```